File: app/src/main/java/reschikov/geekbrains/androidAdvancedLevel/weatherApplication/reply/FragmentAdapter.java
package reschikov.geekbrains.androidadvancedlevel.weatherapplication.reply;

import android.content.Context;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import java.util.ArrayList;
import reschikov.geekbrains.androidadvancedlevel.weatherapplication.R;

class FragmentAdapter extends FragmentPagerAdapter {

    private final ArrayList<Fragment> listFragment;
    private final Context context;
    private final FragmentManager fm;

    FragmentAdapter(FragmentManager fm, ArrayList<Fragment> listFragment, Context context) {
        super(fm);
        this.fm = fm;
        this.listFragment = listFragment;
        this.context = context;
    }

    public void clear(){
        for (int i = 0; i <listFragment.size() ; i++) {
            fm.beginTransaction().remove(listFragment.get(i)).commitNowAllowingStateLoss();
        }
    }

    @Override
    public Fragment getItem(int i) {
        return listFragment.get(i);
    }

    @Override
    public int getCount() {
        if (listFragment == null) return 0;
        return listFragment.size();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        if (position == 0) return context.getString(R.string.current_state);
        return context.getString(R.string.forecast);
    }
}

File: app/src/main/java/reschikov/geekbrains/androidAdvancedLevel/weatherApplication/ui/weather/FragmentAdapter.kt
package reschikov.geekbrains.androidadvancedlevel.weatherapplication.ui.weather

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

internal class FragmentAdapter(fm: FragmentManager,
    private val listFragment: List<Pair<String, Fragment>>)
        : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(i: Int): Fragment {
        return listFragment[i].second
    }

    override fun getCount(): Int {
        return listFragment.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return listFragment[position].first
    }

    fun findFragment(title: String): Fragment? {
        return listFragment.find {it.first == title}?.second
    }
}
